lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -1260,9 +1260,16 @@ bool OverrideMatcher::checkOverride(ValueDecl *baseDecl,
     auto propertyTy = property->getInterfaceType();
     auto parentPropertyTy = getSuperMemberDeclType(baseDecl);
 
+    // If @preconcurrency, strip concurrency from decl before matching
+    if (baseDecl->preconcurrency() && !baseDecl->isObjC()){
+        attempt = OverrideCheckingAttempt::MismatchedSendability;
```

Comment:
If you want to factor in `attempt` here it should be through a comparison i.e. `attempt == OverrideCheckingAttempt::MismatchedSendability` but it shouldn't be re-assigned because the method is driven by some other code that gives it `attempt`.

Reply:
Ah the other code you're talking about is in `OverriddenDeclsRequest` which always starts by setting attempt to be `PerfectMatch`. But in the case  where we have an overridden `var`(not `func`) with `Sendable` attribute, we want to skip PerfectMatch attempt and skip straight to checking for `MismatchedSendability`

Reply:
I don't think we need to skip anything since perfect match is going to fail anyway but even if we did this is not the way do that, it would have to be done by the caller.